distribution/pull_v2.go

```diff
@@ -147,6 +149,10 @@ func (ld *v2LayerDescriptor) DiffID() (layer.DiffID, error) {
 	return ld.V2MetadataService.GetDiffID(ld.digest)
 }
 
+func (ld *v2LayerDescriptor) ForeignSource() *distribution.Descriptor {
+	return ld.foreignSrc
```

Comment:
Great!

See https://github.com/docker/docker/pull/22866#discussion_r64658849. I only think we need to ensure that we don't have to do too much work to get this `Descriptors` working for other layers.